designer/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentService.scala

```diff
@@ -151,6 +152,23 @@ class DeploymentService(
     } yield ()
   }
 
+  private def checkActiveScenariosLimits(
+      scenario: ScenarioWithDetailsEntity[CanonicalProcess],
+      deployer: LoggedUser,
+      deploymentUpdateStrategy: DeploymentUpdateStrategy,
+  ) = {
+    implicit val loggedUser: LoggedUser = deployer
+    // note: we use "unsafe" version here (without critical section), because we're already in the actions critical section,
```

Comment:
... if we assume that nobody will spawn two deployments via both api concurrently :)

Reply:
I thought that the critical section is used in both deployment services. Tests showed me that it's not. I refactored the code a little bit, and now the "unsafe" method is not needed.